packages/material-ui/src/Radio/Radio.js

```diff
@@ -58,6 +58,15 @@ const RadioRoot = styled(SwitchBase, {
   },
 }));
 
+function areEqualValues(a, b) {
+  if (typeof b === 'object' && b !== null) {
+    return a === b;
+  }
+
+  // The value could be a number, the DOM serialize it
+  return String(a) === String(b);
```

Comment:
If this change is just about number then let's reverse this comparison by only stringifying `typeof b === 'number'` and use strict value comparison otherwise. This avoids unrelated changes in behavior (e.g. symbols, `undefined`, boolean etc).

Reply:
Makes more sense to implement consistent behavior. It's selecting from a list of values. Select and RadioGroup is more about representation than behavior as far as I can tell so consistent behavior seems more important.